### PR TITLE
Add `background` parameter to `Screen`

### PR DESCRIPTION
**Pull request by elParaguayo**
Adds a new `background` parameter which will paint a solid colour the screen.

### CODE REVIEW

File: libqtile/core/manager.py [hunk 1]
```diff
@@ -450,6 +460,15 @@ def reconfigure_screens(self, *_: list[Any], **__: dict[Any, Any]) -> None:
...
         self.core.painter.paint(screen, image_path, mode)
 
+    def fill_screen(self, screen: Screen, background: ColorType) -> None:
+        with tempfile.NamedTemporaryFile() as temp:
+            surface = cairocffi.ImageSurface(cairocffi.FORMAT_ARGB32, screen.width, screen.height)
+            with cairocffi.Context(surface) as ctx:
+                ctx.set_source_rgba(*rgb(background))
+                ctx.paint()
+            surface.write_to_png(temp.name)
+            self.core.painter.paint(screen, temp.name)
```
Comment (tych0) [inline libqtile/core/manager.py:470]: Can we add a method to get the background surface and do this painting via:

```
ctx.set_source_rgba(0, 0, 0, 0.5)
ctx.rectangle(0, 0, WIDTH, HEIGHT)
ctx.fill()
```
?

it seems a little ugly to me to write a temp file to do this.

Comment (elParaguayo) [reply libqtile/core/manager.py:470]: Sure. I didn't want to create a temporary file. I just did this as it allows us to use the `paint` method. The advantage of that was to use the existing code for updating the relevant atoms in x11.

I was being a bit lazy so let me rework this and see if I can refactor those methods.

**Comment by elParaguayo** [comment]
OK - I've refactored this so we're not using a temporary file.

**Comment by elParaguayo** [comment]
Don't merge this yet - I suspect there's a more efficient way to do this on Wayland.

**Comment by elParaguayo** [comment]
@jwijenbergh Can you have a look at the wayland implementation here please.

**Review by jwijenbergh — APPROVED** [review]
No comments here for wayland, works perfectly and code is clean

**Comment by elParaguayo** [comment]
Thanks. I'll rebase this into a single commit then.

**Pull request merged**